### PR TITLE
Reduce  class complexity #148

### PR DESCRIPTION
**Pull request by elminson**
I just send a PR with function `getArrayLog` reusing the log array creation and `setFileAll` to make sure $file is set

### CODE REVIEW

**Comment by elminson** [comment]
@rap2hpoutre  any comments on this?

File: src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php [hunk 1]
```diff
@@ -174,28 +184,42 @@ public function all()
...
+    /**
+     * Create array data from log
+     * @param $index
+     * @param $current
+     * @param $level
+     * @param $key
+     * @param $line
+     * @return array
+     */
+    protected function getArrayLog($index, $current, $level, $key, $line)
```
Comment (rap2hpoutre) [inline src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php]: I'm not totally sure this function makes the code more readable (but maybe I'm wrong). I mean it adds a bunch of ternary operators (7 ternary operator) and is more complex to read. It avoid repetition of keys, good point, but it adds some complexity IMO. And there is a new function with _5_ params which are not really simple to understand TBH. I don't _actually_ think it's more readable, but I understand your idea. What are your thoughts?

**Comment by elminson** [comment]
Thank you for your comments, I will check all and let you know.

File: src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php [hunk 2]
```diff
@@ -33,6 +33,11 @@ class LaravelLogViewer
      */
     private $level;
 
+    /**
+     * @var Level level
```
[the remark this answers is not in the document]
Comment (elminson) [reply src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php]: Yes you are right  :)

File: src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php [hunk 3]
```diff
@@ -174,28 +184,42 @@ public function all()
...
-                    'context' => '',
-                    'level' => '',
-                    'level_class' => '',
-                    'level_img' => '',
-                    'date' => $key + 1,
-                    'text' => $line,
-                    'in_file' => null,
-                    'stack' => '',
-                ];
+                $log[] = $this->getArrayLog('', '', '', $key, $line);
```
[the remark this answers is not in the document]
Comment (elminson) [reply src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php]: I miss that one :) good catch!

File: src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php [hunk 4]
```diff
@@ -174,28 +184,42 @@ public function all()
...
+    {
+        return array(
+            'context' => isset($current[3]) ? $current[3] : '',
+            'level' => isset($level) ? $level : '',
+            'level_class' => isset($level) ? $this->level->cssClass($level) : '',
+            'level_img' => isset($level) ? $this->level->img($level) : '',
+            'date' => isset($current[1]) ? $current[1] : $key + 1,
+            'text' => isset($current[4]) ? $current[4] : $line,
+            'in_file' => isset($current[5]) ? $current[5] : null,
+            'stack' => preg_replace("/^\n*/", '', $this->log_data[$index])
```
[the remark this answers is not in the document]
Comment (elminson) [reply src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php]: No, preg_replace if $this->log_data[$index] return empty will work perfect. 
BTW we need the test cases ASAP, maybe we can work  on that this weekend :)

File: src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php [hunk 5]
```diff
@@ -221,7 +245,8 @@ public function getFolderFiles($basename = false)
     public function getFiles($basename = false, $folder = '')
     {
         $pattern = function_exists('config') ? config('logviewer.pattern', '*.log') : '*.log';
-        $files = glob($this->storage_path.'/' . $folder . '/' . $pattern, preg_match($this->pattern->getPattern('files'), $pattern) ? GLOB_BRACE : 0);
+        $files = glob($this->storage_path . '/' . $folder . '/' . $pattern,
```
[the remark this answers is not in the document]
Comment (elminson) [reply src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php]: I like your approach :)

File: src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php [hunk 6]
```diff
@@ -231,4 +256,4 @@ public function getFiles($basename = false, $folder = '')
         }
         return array_values($files);
     }
-}
+}
```
[the remark this answers is not in the document]
Comment (elminson) [reply src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php]: cool ;)

File: src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php (hunk 1 above)
Comment (elminson) [reply src/Rap2hpoutre/LaravelLogViewer/LaravelLogViewer.php]: What about this approach
```php
$log[] = $this->getArrayLog([
                    'index' => '',
                    'current' => [],
                    'level' => '',
                    'key' => $key,
                    'line' => $line
                ]);
```
And the function 
```php
/**
     * Create array data from log
     * @param $data
     * @return array
     */
    protected function getArrayLog($data)
    {
        $index = $data['index'];
        $current = $data['current'];
        $level = $data['level'];
        $key = $data['key'];
        $line = $data['line'];
```

**Comment by rap2hpoutre** [comment]
@elminson Hi, and thank you for your contribution 😄 !

I just checked and I'm sorry I can't "approve" the actual PR 😢. I think we should try to avoid mixing different enhancements/fix in a same PR: the risk is it will be never accepted by the reviewer, since every single line could lead to a discussion. For example, where I use to work (we use PR system too, with code review, etc.), we try to make PR small as possible, with only one concern. Not sure it's the right way, but it seems to work for years. A ~big PR addressing various issues _could_ be merged, but it's risky to submit such a PR because it could takes a long time to be accepted. 

In this particular case, as a reviewer I start to be a bit lost: there are some comments that are not answered, then new commit about a new subject. Be sure I still appreciate your work.

**Comment by elminson** [comment]
@rap2hpoutre I understand, I will recreate the PR and separate enhancements. 
Probably will get the PR in the next 2 days

**Comment by rap2hpoutre** [comment]
@elminson Thank you!! About reducing class complexity, I feel like I must think by my side first about what I want. I will try to improve it a bit myself before accepting PR. Thank you for the other PR ;)

**Pull request closed without merging**